libs/orchestrator/json_models.go

```diff
@@ -37,37 +43,41 @@ func (j *JobJson) IsApply() bool {
 
 func JobToJson(job Job) JobJson {
 	return JobJson{
-		ProjectName:       job.ProjectName,
-		ProjectDir:        job.ProjectDir,
-		ProjectWorkspace:  job.ProjectWorkspace,
-		Terragrunt:        job.Terragrunt,
-		Commands:          job.Commands,
-		ApplyStage:        stageToJson(job.ApplyStage),
-		PlanStage:         stageToJson(job.PlanStage),
-		PullRequestNumber: job.PullRequestNumber,
-		EventName:         job.EventName,
-		RequestedBy:       job.RequestedBy,
-		Namespace:         job.Namespace,
-		StateEnvVars:      job.StateEnvVars,
-		CommandEnvVars:    job.CommandEnvVars,
+		ProjectName:        job.ProjectName,
+		ProjectDir:         job.ProjectDir,
+		ProjectWorkspace:   job.ProjectWorkspace,
+		Terragrunt:         job.Terragrunt,
+		Commands:           job.Commands,
+		ApplyStage:         stageToJson(job.ApplyStage),
+		PlanStage:          stageToJson(job.PlanStage),
+		PullRequestNumber:  job.PullRequestNumber,
+		EventName:          job.EventName,
+		RequestedBy:        job.RequestedBy,
+		Namespace:          job.Namespace,
+		StateEnvVars:       job.StateEnvVars,
+		CommandEnvVars:     job.CommandEnvVars,
+		StateEnvProvider:   job.StateEnvProvider,
```

Comment:
So by the time we get here we will have StateEnvRole and CommandEnvRole as strings and we need to convert them to providers using `orchestrator.GetProviderFromRole()` but also check for "" in that function and return nil if so

Reply:
Slightly confused about this point. We're storing them as strings but the return value of the `orchestrator.GetProviderFromRole()` is a `*stscreds.WebIdentityRoleProvider`?